doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,258 @@
+# (footprint).
+#
+# Here, we use the ``diamond`` function from the ``morphology`` module to
+# create a diamond-shaped footprint.
+# An erosion followed by a dilation is called an *opening*.
+# First, we apply an opening filter, in order to remove small objects and thin
+# lines, while preserving the shape and size of larger objects.
+
+footprint = ski.morphology.diamond(3)
+mask_open = ski.morphology.opening(mask_2, footprint)
+plot_comparison(mask_2, mask_open, "mask before", "after opening")
+
+#####################################################################
+# Since "opening" an image starts with an erosion operation, bright regions
+# which are smaller than the structuring element have been removed.
+# When applying an opening filter, tweaking the footprint parameter lets us
+# control how fine-grained the removed features are. For example, if we used
+# ``footprint = ski.morphology.diamond(1)`` in the above, we could see that
+# only smaller features would be filtered out, hence retaining more spots in
+# the mask.
+# Next, we can make the detected areas wider by applying a dilation filter:
+
+mask_dilate = ski.morphology.dilation(mask_open, footprint)
+plot_comparison(mask_open, mask_dilate, "before", "after dilation")
+
+#####################################################################
+# Dilation enlarges bright regions and shrinks dark regions.
+# Notice how, indeed, the white spots have thickened.
+
+#####################################################################
+# Inpaint each frame separately
+# =============================
+# We are now ready to apply inpainting to each frame. For this we use function
+# ``inpaint_biharmonic`` from the ``restoration`` module. It implements an
+# algorithm based on biharmonic equations.
+# This function takes two arrays as inputs:
+# The image to restore and a mask (with same shape) corresponding to the
+# regions we want to inpaint.
+
+image_seq_inpainted = np.zeros(image_seq.shape)
+
+for i in range(image_seq.shape[0]):
+    image_seq_inpainted[i] = ski.restoration.inpaint_biharmonic(
+        image_seq[i],
+        mask_dilate
+    )
+
+#####################################################################
+# Let us visualize one restored image, where the dirt spots have been
+# inpainted. First, we find the contours of the dirt spots (well, of the mask)
+# so we can draw them on top of the restored image:
+
+contours = ski.measure.find_contours(mask_dilate)
+
+# Gather all (row, column) coordinates of the contours
+x = []
+y = []
+for contour in contours:
+    x.append(contour[:, 0])
+    y.append(contour[:, 1])
+# Flatten them
+x_flat = np.concatenate(x).ravel()
+y_flat = np.concatenate(y).ravel()
+# Create a sparse matrix corresponding to the mask of these contours
+data = np.ones(x_flat.shape, dtype='bool')
+mtx = sparse.coo_matrix((data, (x_flat, y_flat)), shape=mask_dilate.shape)
+# Convert it to array
+arr = mtx.toarray().astype('int')
```

Comment:
Cool! I learned something new. Is the fact that `x_flat` & `y_flat` contain floats (often `.5` or `.3`) intentionally ignored? It's not clarified in the [`coo_matrix` docstring](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.coo_matrix.html#scipy.sparse.coo_matrix), local testing and the [source code](https://github.com/scipy/scipy/blob/2e16ec5bc62b7b7db21b37b838fc771c7d232006/scipy/sparse/_coo.py#L164-L165) indicate that digits after the decimal point are just dropped. So wouldn't explicit rounding be "best practice" here?

Also,
```suggestion
x_flat = np.concatenate(x).ravel().round().astype(int)
y_flat = np.concatenate(y).ravel().round().astype(int)
# Create mask of these contours
arr = np.zeros(x_flat.shape, dtype='bool')
arr[x_flat, y_flat] = 1
```
seems simpler to me and avoids the detour in sparse array territory?

Also, I think `arr` is a bad variable name. How about `contour_mask`?

Reply:
> Is the fact that x_flat & y_flat contain floats (often .5 or .3) intentionally ignored?

It's expected (see docstring and [Examples](https://scikit-image.org/docs/stable/api/skimage.measure.html#skimage.measure.find_contours)) but, yes, considering how we are using the computed contours, we should handle this explicitly.

> So wouldn't explicit rounding be "best practice" here?

Right.

Reply:
> Also, I think `arr` is a bad variable name. How about `contour_mask`?

Yes, I agree. My bad for copying-pasting from my {draft, WIP, exploratory} notebook!

Reply:
> seems simpler to me and avoids the detour in sparse array territory?

You're absolutely right! Thanks.